### ISSUE TITLE
Error multithread compile

### ISSUE THREAD

**Issue by platinumthinker** [rebar/rebar#569]
I see sometimes this error:
```
src/commands/cpt_declare_trace_cmd.erl:none: internal error in save_abstract_code;
crash reason: {badmatch,{error,{already_started,<0.391.0>}}}

  in function  beam_lib:call_crypto_server_1/1 (beam_lib.erl, line 907)
  in call from compile:abstract_code/1 (compile.erl, line 1256)
  in call from compile:save_abstract_code/1 (compile.erl, line 1236)
  in call from compile:'-internal_comp/4-anonymous-1-'/2 (compile.erl, line 292)
  in call from compile:fold_comp/3 (compile.erl, line 310)
  in call from compile:internal_comp/4 (compile.erl, line 294)
  in call from compile:'-do_compile/2-anonymous-0-'/2 (compile.erl, line 153)
Compiling src/commands/cpt_declare_trace_cmd.erl failed:
ERROR: compile failed while processing /home/thinker/develop/repos/all/deps/ecss_cpt_server: rebar_abort
```
But i run rebar in one thread (rebar co -j1) this works...

**Comment by lrascao** [comment]
Do you have a public repo available where this happens? Also what OTP version are you using?

**Comment by platinumthinker** [comment]
I use Erlang/OTP 17 [erts-6.4.1.2] and lastest rebar from master

**Comment by ferd** [comment]
for what it's worth, I believe we entirely dropped parallel compilation in rebar3 and noticed no ill-effect on overall compilation speed.

**Comment by lrascao** [comment]
@platinumthinker do you see this happening on a public repo where i could try it out?

**Comment by tuncer** [comment]
@platinumthinker why did you close this? Can't you reproduce it anymore? If you can reproduce it, did the bug appear after a specific rebar release?

**Comment by platinumthinker** [comment]
@tuncer this bug reproduced on several repository but i not retry 100%... This bug appears once per week =(

**Comment by platinumthinker** [comment]
@ferd thank you for true about not effective parallel compilation (i use core i7 and SSD)

**Comment by lrascao** [comment]
@platinumthinker can you please point some of those repositories where you found the issue?

**Comment by lrascao** [comment]
This is actually pretty easy to reproduce:
`[spawn(fun() -> beam_lib:get_crypto_key(<<"ola">>) end) || _ <- lists:seq(0,10)].`
`beam_lib:call_crypto_server_1` is not checking for an already started crypto server, `compile:abstract_code` calls `beam_lib:get_crypto_key`, since there might be several processes
compiling code this race condition becomes apparent. A fix has been proposed for OTP (erlang/otp#921)

**Comment by platinumthinker** [comment]
@lrascao Thanks! This good patch solved my problem
